src/structures/MessageComponentInteraction.js

```diff
@@ -51,6 +51,18 @@ class MessageComponentInteraction extends Interaction {
     this.webhook = new InteractionWebhook(this.client, this.applicationID, this.token);
   }
 
+  /**
+   * The component which was interacted with
+   * @type {?(MessageActionRowComponent|Object)}
+   * @readonly
+   */
+  get component() {
+    return this.message.components
+      .map(row => row.components)
+      .flat()
+      .find(component => (component.customID ?? component.custom_id) === this.customID) ?? null;
```

Comment:
This should fix ESLint maybe?
```suggestion
    return (
      this.message.components
        .map(row => row.components)
        .flat()
        .find(component => (component.customID ?? component.custom_id) === this.customID) ?? null
    );
```

Reply:
`?? null` should be used on the return value of `Array#find` here to ensure that the getter returns `null` and not `undefined` if the component wasn't found.

Moving it into the predicate callback does nothing as `boolean ?? null` will never return `null` (and `false` is perfectly fine to return there).

Reply:
@DTrombett Trying to manually fix ESLint is not what should happen here, we have hooks for that for a reason.